### PR TITLE
Add the possibility to duplicate a case when creating a study

### PR DESCRIPTION
**Pull request by flomillot**
Signed-off-by: Florent MILLOT <millotflo@gmail.com>

### CODE REVIEW

File: src/main/java/org/gridsuite/explore/server/ExploreController.java [hunk 1]
```diff
@@ -69,12 +70,12 @@ public ResponseEntity<Void> createStudy(@PathVariable("studyName") String studyN
...
-                                            @RequestParam("studyName") String studyName,
-                                            @RequestParam("description") String description,
-                                            @RequestParam(QUERY_PARAM_PARENT_DIRECTORY_ID) UUID parentDirectoryUuid,
-                                            @RequestHeader("userId") String userId) {
-        exploreService.createStudy(sourceStudyUuid, studyName, description, userId, parentDirectoryUuid);
+    public ResponseEntity<Void> duplicateStudy(@RequestParam("duplicateFrom") UUID sourceStudyUuid,
+                                               @RequestParam("studyName") String studyName,
+                                               @RequestParam("description") String description,
+                                               @RequestParam(QUERY_PARAM_PARENT_DIRECTORY_ID) UUID parentDirectoryUuid,
+                                               @RequestHeader("userId") String userId) {
```
Comment (bendaamerahm) [inline src/main/java/org/gridsuite/explore/server/ExploreController.java:77]: remove identation changes

Comment (flomillot) [reply src/main/java/org/gridsuite/explore/server/ExploreController.java:77]: The name of the method is longer, so it's normal that the formatter changed the indentation of the params no ?

**Review by bendaamerahm — APPROVED** [review]

File: src/main/java/org/gridsuite/explore/server/services/ExploreService.java [hunk 2]
```diff
@@ -54,23 +54,16 @@ public ExploreService(
         this.caseService = caseService;
     }
 
-    public void createStudy(String studyName, UUID caseUuid, String description, String userId, UUID parentDirectoryUuid, Map<String, Object> importParams) {
+    public void createStudy(String studyName, UUID caseUuid, String description, String userId, UUID parentDirectoryUuid, Map<String, Object> importParams, Boolean duplicateCase) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/explore/server/services/ExploreService.java:57]: Use primitive type

**Review by SlimaneAmar — APPROVED** [review]

File: src/main/java/org/gridsuite/explore/server/ExploreController.java [hunk 3]
```diff
@@ -46,11 +46,12 @@ public ExploreController(ExploreService exploreService, DirectoryService directo
     @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "Study creation request delegated to study server")})
     public ResponseEntity<Void> createStudyFromExistingCase(@PathVariable("studyName") String studyName,
```
[the remark this answers is not in the document]
Comment (flomillot) [reply src/main/java/org/gridsuite/explore/server/ExploreController.java]: Done

File: src/main/java/org/gridsuite/explore/server/services/ExploreService.java (hunk 2 above)
Comment (flomillot) [reply src/main/java/org/gridsuite/explore/server/services/ExploreService.java:57]: Was done in commit https://github.com/gridsuite/explore-server/pull/42/commits/d16af1720af9c107ce5efbed7adccf1f634c4bd0

**Pull request merged**